cmd/sonicd/metrics/disksize.go

```diff
@@ -14,17 +15,20 @@ var once sync.Once
 
 func SetDataDir(datadir string) {
 	once.Do(func() {
-		go measureDbDir("db_size", datadir)
-		go measureDbDir("statedb/disksize", filepath.Join(datadir, "carmen"))
+		go measureDbDir(context.Background(), "db_size", datadir)
+		go measureDbDir(context.Background(), "statedb/disksize", filepath.Join(datadir, "carmen"))
 	})
 }
 
-func measureDbDir(name, datadir string) {
+func measureDbDir(ctx context.Context, name, datadir string) {
 	var (
-		gauge = metrics.GetOrRegisterGauge(name, nil)
+		gauge  = metrics.GetOrRegisterGauge(name, nil)
 		rescan = len(datadir) > 0 && datadir != "inmemory"
 	)
 	for rescan {
+		if ctx.Err() != nil {
+			return
+		}
```

Comment:
If the goal is to support the cancellation of this function by canceling the context, the sleep should be replaced by a select like this (not tested, just a rough outline):
```
select {
  case <-ctx.Done(): return
  case <-time.After(time.Minute):
}
```

Reply:
Also, unrelated to your changes: Since this is a monitoring utility that wants to sample the disk usage periodically, a timer should be used instead of a sleep; this way, you can avoid the drifting of the sampling due to the delay introduced by measuring the disk size;